### PR TITLE
add scoped to all ncappnavigation components

### PR DESCRIPTION
**Pull request by szaimen**
Close https://github.com/nextcloud/server/issues/41153

### CODE REVIEW

**Comment by szaimen** [comment]
This might have broken NcAppNavigationNewItem:
image

**Comment by szaimen** [comment]
@ShGKme I suppose this is not the fix to the linked problem? Am I missing something?

**Comment by ShGKme** [comment]
> @ShGKme I suppose this is not the fix to the linked problem? Am I missing something?

It should work if all `nextcloud/vue` on a page is updated and has `scoped`.

Is there another app that adds these styles on the page?

Scoping styles with `scoped` is only a one-way scoping, meaning it saves this component from changing other elements on the page but doesn't save this component from being overridden by styles outside.
(+1 for reliable `<style module>`)

**Review by raimund-schluessler — CHANGES_REQUESTED** [review]
> This might have broken NcAppNavigationNewItem:

This is still an issue. Since the styles are now scoped, they need to be duplicated for `NcAppNavigationNewItem` to apply there.

**Comment by susnux** [comment]
Maybe share the styles as in a scss file and import as scoped styles in the components?

File: src/components/NcAppNavigationItem/NcAppNavigationItem.vue [hunk 1]
```diff
@@ -771,208 +771,6 @@ export default {
...
-// Add more contrast for active entry
-[data-themes*='highcontrast'] {
-	.app-navigation-entry {
-		&:active {
-			background-color: var(--color-primary-element-light-hover) !important;
-		}
-	}
-}
+<style scoped lang="scss">
+@import '../../assets/NcAppNavigationItem';
```
Comment (susnux) [inline src/components/NcAppNavigationItem/NcAppNavigationItem.vue:775]: `@import` is deprecated you should use `@use`

Comment (szaimen) [reply src/components/NcAppNavigationItem/NcAppNavigationItem.vue]: done

Comment (szaimen) [reply src/components/NcAppNavigationItem/NcAppNavigationItem.vue:775]: I tried this and it didnt work: https://github.com/nextcloud-libraries/nextcloud-vue/actions/runs/6734727700/job/18306263029

**Review by susnux — APPROVED** [review]

**Review by JuliaKirschenheuter — APPROVED** [review]
works!

**Pull request merged**